back/src/Guide/Api/Types.hs

```diff
@@ -433,19 +433,32 @@ instance ToSchema ItemSection where
 --
 -- When updating it, don't forget to also update 'setItemInfo'.
 data CItemInfo = CItemInfo
-  { ciiUid     :: Uid Item   ? "Item ID"
-  , ciiName    :: Text       ? "Item name"
-  , ciiCreated :: UTCTime    ? "When the item was created"
-  , ciiGroup   :: Maybe Text ? "Item group"
-  , ciiHackage :: Maybe Text ? "Package name on Hackage"
-  , ciiLink    :: Maybe Url  ? "Link to the official site, if exists"
+  { ciiName    :: Maybe Text         ? "Item name"
+  , ciiGroup   :: Maybe (Maybe Text) ? "Item group"
+  , ciiHackage :: Maybe (Maybe Text) ? "Package name on Hackage"
+  , ciiLink    :: Maybe (Maybe Url)  ? "Link to the official site, if exists"
   } deriving (Show, Generic)
 
 instance A.ToJSON CItemInfo where
-  toJSON = A.genericToJSON jsonOptions
+  toJSON cii = A.object
```

Comment:
Almost right.

If the field is `Nothing`, it should be omitted entirely. With this instance, all fields will always be present.

What you want is something like this:

```haskell
object $ catMaybes
  [ ("name" A..=) <$> unH (ciiName cii) 
  , ("group" A..=) <$> unH (ciiGroup cii)
  , ...
  ]
```